services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java

```diff
@@ -402,7 +402,17 @@ public void publishNews(News newNews, String publisher) throws Exception {
     List<String> oldTargets = newsTargetingService.getTargetsByNewsId(newNews.getId());
     if(newNews.getTargets() != null && !oldTargets.equals(newNews.getTargets())) {
       newsTargetingService.deleteNewsTargets(newNews.getId());
-      newsTargetingService.saveNewsTarget(newNews.getId(), newNews.getTargets(), publisher);
+      newsTargetingService.saveNewsTarget(newNews.getId(), StringUtils.equals(news.getPublicationState(), PublicationDefaultStates.STAGED), newNews.getTargets(), publisher);
+    }
+    //Update staged property to false when news is published.
+    Map<String, String> properties = new LinkedHashMap<>();
+    properties.put(PublicationDefaultStates.STAGED, String.valueOf(false));
+    List<MetadataItem> targets = newsTargetingService.getTargetsListByNewsId(newNews.getId());
+    if(!Objects.isNull(targets)) {
+      for (MetadataItem metadataItem : targets) {
+        metadataItem.setProperties(properties);
+        newsTargetingService.updateNewsTarget(metadataItem);
+      }
```

Comment:
I think that you don't need this, you should just remove the above condition if(newNews.getTargets() != null && !oldTargets.equals(newNews.getTargets())) to delete and save news targets. You need that also to refresh the publish date